web/src/lib/domain-logic/allowFormation.ts

```diff
@@ -1,8 +1,29 @@
-import { PublicFilingLegalType, publicFilingLegalTypes } from "@businessnjgovnavigator/shared";
+import {
+  BusinessPersona,
+  castPublicFilingLegalTypeToFormationType,
+  FormationLegalType,
+  PublicFilingLegalType,
+  publicFilingLegalTypes,
+} from "@businessnjgovnavigator/shared";
+
+export const allowFormation = (
+  legalStructureId: string | undefined,
+  persona: BusinessPersona | undefined
+): boolean => {
+  const featureFlagMap: Partial<Record<FormationLegalType, boolean>> = {
+    "foreign-limited-partnership": process.env.FEATURE_BUSINESS_FLP === "true",
+  };
 
-export const allowFormation = (legalStructureId: string | undefined): boolean => {
   if (publicFilingLegalTypes.includes(legalStructureId as PublicFilingLegalType)) {
-    return true;
+    if (persona === "FOREIGN" || legalStructureId === "nonprofit") {
```

Comment:
[sand] nonprofits are no longer covered here. We can safely remove this portion of the check.
```suggestion
    if (persona === "FOREIGN") {
```